wgpu_shadertoy/shadertoy.py

```diff
@@ -240,9 +241,12 @@ def _prepare_canvas(self):
         self._present_context: wgpu.GPUCanvasContext = self._canvas.get_context()
 
         # We use "bgra8unorm" not "bgra8unorm-srgb" here because we want to let the shader fully control the color-space.
+        # broken in newer versions of wgpu-py it seems... due to the minimal Vulkan capabilities...
         # TODO: instead use canvas preference? ref: GPUCanvasContext.get_preferred_format()
+        self._format = self._present_context.get_preferred_format(adapter=self._device.adapter)
+
         self._present_context.configure(
-            device=self._device, format=wgpu.TextureFormat.bgra8unorm
+            device=self._device, format=self._format
```

Comment:
Can also set to `None` to have it select the preferred format. Less code, unless you need `self._format`. Note that the format is also accessible in `texture.format` on the texture obtained via `get_current_texture()`.

Reply:
hm, I had a look and the only other solution I can think of is to make it a property of the `ImageRenderPass`, as it's needed to create the render pipeline. The awful part is that at init time for the Image class, the canvas(`_present_context`) might not be accessible via the parent instance of `Shadertoy`.
It could be returned by this method instead of passed via an attribute. It could be useful to have available when translating the snapshot back into RGB.